addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2587,6 +2588,35 @@ options.registry.ScrollButton = options.Class.extend({
     },
 });
 
+/**
+ * Allows for images to be replaced.
+ */
+options.registry.ReplaceImage = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        var $button = this.$el.find('we-button');
+        var $overlayArea = this.$overlay.find('.oe_snippet_remove');
+        $button.insertBefore($overlayArea);
+
+        return this._super(...arguments);
+    },
+    
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Replaces the image.
+     *
+     * @see this.selectClass for parameters
+     */
+    replaceImage: function () {
+        new weWidgets.MediaDialog(this, {}, this.$target).open();
+    },
```

Comment:
Method options should return a Promise that indicates when the option is done doing its work. In this case, it is only once the `MediaDialog` has been closed, so you have to wait for that. This is to avoid inconsistencies: two option methods called in a certain order should be fully executed in that order to not potentially mixing them in a random way.

Also, are you sure there is not supposed to be any option given to that media dialog initialization? There should be the same options as when you double click on that image, and there is probably code to share to do that. Tell me if I am wrong, it is just a guess.

Reply:
I have initialized the media dialog the same way as it is done in summernote for the double click. I do not know if it is possible or desirable to refactor that initialization somewhere else, as it involves changing the summernote library, but I would have done that directly in the media dialog class, allowing to init it directly with the $editable. What do you think ?

Reply:
I see all this for the double click:
```javascript
var media = $(r.sc).parents().addBack().filter(function (i, el) {
        return dom.isImg(el);
    })[0];
    var options = $editable.closest('.o_editable, .note-editor').data('options');
    topBus.trigger('media_dialog_demand', {
        $editable: $editable,
        media: media,
        options: {
            onUpload: $editable.data('callbacks').onUpload,
            noVideos: options && options.noVideos,
        },
        onSave: function (newMedia) {
            if (!newMedia) {
                return;
            }
            if (media) {
                $(media).replaceWith(newMedia);
            } else {
                r.insertNode(newMedia);
            }
        },
    });
```
So it seems like there are options to consider... ;)

But you are right this is annoying the refactor while the edietor team is changing that code entirely. Maybe let's go the easy way and simulate a double click on the image in the options and add a TODO comment to say we'll have to do better once the editor work is merged.